src/main/java/tfb/status/handler/ShareResultsPageHandler.java

```diff
@@ -0,0 +1,168 @@
+package tfb.status.handler;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import io.undertow.server.HttpHandler;
+import io.undertow.server.HttpServerExchange;
+import io.undertow.server.handlers.DisableCacheHandler;
+import io.undertow.server.handlers.form.EagerFormParsingHandler;
+import io.undertow.server.handlers.form.FormData;
+import io.undertow.server.handlers.form.FormDataParser;
+import io.undertow.server.handlers.form.FormParserFactory;
+import io.undertow.server.handlers.form.MultiPartParserDefinition;
+import tfb.status.handler.routing.ExactPath;
+import tfb.status.hk2.extensions.Provides;
+import tfb.status.service.MustacheRenderer;
+import tfb.status.service.ShareResultsUploader;
+import tfb.status.undertow.extensions.HttpHandlers;
+import tfb.status.undertow.extensions.MethodHandler;
+import tfb.status.view.ShareResultsJsonView;
+import tfb.status.view.ShareResultsPageView;
+
+import static com.google.common.net.MediaType.HTML_UTF_8;
+import static io.undertow.util.Headers.CONTENT_TYPE;
+import static io.undertow.util.Methods.GET;
+import static io.undertow.util.Methods.POST;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * The standalone paste bin style page for sharing results files.
+ * Allows anonymous users to upload a results.json file or paste in
+ * the contents of a results.json file in order to share it.
+ *
+ * @see ShareResultsUploader
+ */
+@Singleton
+public class ShareResultsPageHandler implements HttpHandler {
+  private final MustacheRenderer mustacheRenderer;
+  private final ShareResultsUploader shareResultsUploader;
+
+  @Inject
+  public ShareResultsPageHandler(MustacheRenderer mustacheRenderer,
+                                 ShareResultsUploader shareResultsUploader) {
+    this.mustacheRenderer = Objects.requireNonNull(mustacheRenderer);
+    this.shareResultsUploader = Objects.requireNonNull(shareResultsUploader);
+  }
+
+  @Provides
+  @Singleton
+  @ExactPath("/share-results-pastebin")
+  public HttpHandler shareResultsPageHandler() {
+    FormParserFactory formParserFactory = FormParserFactory.builder()
+        .addParsers(new MultiPartParserDefinition())
+        .build();
+
+    return HttpHandlers.chain(
+        this,
+        handler -> new MethodHandler()
+            .addMethod(GET, handler)
+            .addMethod(POST,
+                new EagerFormParsingHandler(formParserFactory).setNext(handler)),
+        handler -> new DisableCacheHandler(handler));
+  }
+
+  @Override
+  public void handleRequest(HttpServerExchange exchange) throws Exception {
```

Comment:
Please narrow the exception type of this method if possible.  Can it be declared to only throw `IOException`?